salt/modules/boto_lambda.py

```diff
@@ -0,0 +1,783 @@
+
+log = logging.getLogger(__name__)
+
+# Import third party libs
+import salt.ext.six as six
+# pylint: disable=import-error
+try:
+    #pylint: disable=unused-import
+    import boto
+    import boto3
+    #pylint: enable=unused-import
+    from botocore.exceptions import ClientError
+    logging.getLogger('boto').setLevel(logging.CRITICAL)
+    logging.getLogger('boto3').setLevel(logging.CRITICAL)
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+# pylint: enable=import-error
+
+
+def __virtual__():
+    '''
+    Only load if boto libraries exist and if boto libraries are greater than
+    a given version.
+    '''
+    required_boto_version = '2.8.0'
+    required_boto3_version = '1.2.1'
+    # the boto_lambda execution module relies on the connect_to_region() method
+    # which was added in boto 2.8.0
+    # https://github.com/boto/boto/commit/33ac26b416fbb48a60602542b4ce15dcc7029f12
+    if not HAS_BOTO:
+        return False
+    elif _LooseVersion(boto.__version__) < _LooseVersion(required_boto_version):
+        return False
+    elif _LooseVersion(boto3.__version__) < _LooseVersion(required_boto3_version):
+        return False
+    else:
+        return True
+
+
+def __init__(opts):
+    salt.utils.compat.pack_dunder(__name__)
+    if HAS_BOTO:
+        __utils__['boto3.assign_funcs'](__name__, 'lambda')
+
+def _find_function(name,
+               region=None, key=None, keyid=None, profile=None):
+
+    '''
+    Given function name, find and return matching Lambda information.
+    '''
+    conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+
+    for funcs in salt.utils.boto3.paged_call(conn.list_functions):
+        for func in funcs['Functions']:
+            if func['FunctionName'] == name:
+                return func
+    return None
+
+
+def function_exists(FunctionName, region=None, key=None,
+           keyid=None, profile=None):
+    '''
+    Given a function name, check to see if the given function name exists.
+
+    Returns True if the given function exists and returns False if the given
+    function does not exist.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt myminion boto_lambda.function_exists myfunction
+
+    '''
+
+    try:
+        func = _find_function(FunctionName,
+                             region=region, key=key, keyid=keyid, profile=profile)
+        return {'exists': bool(func)}
+    except ClientError as e:
+        return {'error': salt.utils.boto3.get_error(e)}
+
+
+def _get_role_arn(name, region=None, key=None, keyid=None, profile=None):
+    if name.startswith('arn:aws:iam:'):
+        return name
+
+    account_id = __salt__['boto_iam.get_account_id'](
+        region=region, key=key, keyid=keyid, profile=profile
+    )
+    return 'arn:aws:iam::{0}:role/{1}'.format(account_id, name)
+
+
+def _filedata(infile):
+    with open(infile, 'rb') as f:
+       return f.read()
+
+def create_function(FunctionName, Runtime, Role, Handler, ZipFile=None,
```

Comment:
I'm not particularly familiar with Boto, so I'm sure there's a reason to have these kwargs in mixed-case but could you explain more about this decision? I'm wondering if we could normalize them somewhere...